### DISCUSSION TITLE
Download Button from a GitHub release does not work

### DISCUSSION THREAD

**Discussion by hlinero** [hydraulic-software/conveyor#107] [Q&A] [answered]
Hello @mikehearn,

I have successfully published my app through GitHub following [these instructions.](https://conveyor.hydraulic.dev/14.1/configs/download-pages/#publishing-through-github)

The problem is that when I click on the `Download` button from the download site, the link points to this URL:
`https://github.com/{{ORG_NAME}}/{{REPO_NAME}}/latest/download/composeapp-1.0.0-mac-aarch64.zip`

Which results in a `Not Found` page.

I believe the appropriate URL should have been:
`https://github.com/{{ORG_NAME}}/{{REPO_NAME}}/releases/download/1.0.0/composeapp-1.0.0-mac-aarch64.zip` because with that link I can download the `composeapp-1.0.0-mac-aarch64.zip` file.

In the `conveyor.config` file I have set the following:
`base-url = "https://github.com/{{ORG_NAME}}/{{REPO_NAME}}/latest/download"`

I am not sure if this has something to do with the problem or if I am missing something else.

Any ideas of what is causing this? and how to fix it?

Thank you for the help.

**Comment by hlinero** [comment]
@vinceglb Once again, thanks for the help with [this issue](https://github.com/hydraulic-software/conveyor/discussions/97#discussioncomment-9174850)

I was looking at [your project](https://github.com/vinceglb/SpaceDrop) and I noticed that you also have the problem I described above.

If I go to your [download site](https://vinceglb.github.io/SpaceDrop/download) and click on the `Download` button, I also get a `Not Found` Page.

Were you aware of this? Have you had a chance to solve this in any other project?

Thank you

**Reply by vinceglb** [reply]
I found the problem: we are deploying a file called `download.html`. Github pages will just serve these files like this without any redirection.

Here is a correct url to access download page: https://vinceglb.github.io/SpaceDrop/download.html

**Reply by hlinero** [reply]
Yes, but if I go to [https://vinceglb.github.io/SpaceDrop/download.html](https://vinceglb.github.io/SpaceDrop/download.html) and click on the green `Download` button, then I see a black "Not Found page"

![Screenshot 2024-04-23 at 14 06 11](https://github.com/hydraulic-software/conveyor/assets/17732781/2a7f1793-f920-467a-ab57-1cc3c0e52db4)
)

**Reply by hlinero** [reply]
This is the "Not Found" page I was referring to, not the one at [https://vinceglb.github.io/SpaceDrop/](https://vinceglb.github.io/SpaceDrop/) @vinceglb

**Reply by vinceglb** [reply]
In my case, I didn't update my GitHub pages to the latest version.

Here are the source for my Github pages pointing to the 0.1.0 version: https://github.com/vinceglb/SpaceDrop/tree/gh-pages
But my latest version is 0.2.1: https://github.com/vinceglb/SpaceDrop/releases

If I update the link of the green button with the latest version, it's working fine
- https://github.com/vinceglb/SpaceDrop/releases/download/0.2.1/spacedrop-0.2.1-mac-aarch64.zip

**Reply by vinceglb** [reply]
I updated my github pages with the latest version: https://vinceglb.github.io/SpaceDrop/download.html

**Reply by hlinero** [reply]
Do you manually update the link of the green button?

**Reply by vinceglb** [reply]
I have picked the `download.html` file generated by conveyor for the 0.2.1.

All the generated files are in the assets section of the release: https://github.com/vinceglb/SpaceDrop/releases/tag/0.2.1 
I took the `download.html` file and replace it in the `gh-pages` branch.

**Comment by mikehearn** [comment, marked as answer]
> In the conveyor.config file I have set the following: `base-url = "https://github.com/{{ORG_NAME}}/{{REPO_NAME}}/latest/download"`

That's not the correct URL. Please refer to this section of the user guide to learn how to publish to GitHub:

https://conveyor.hydraulic.dev/latest/configs/download-pages/#publishing-through-github

**Reply by hlinero** [reply]
Thanks @mikehearn 

I was writing the incorrect URL. 

Once again thank @vinceglb for your input
